pkg/handlers/host.go

```
@@ -69,12 +92,74 @@ func (h *HostHandlers) GetHostsByTenantID(w http.ResponseWriter, req *http.Reque
+	host, err := h.hostService.PatchHostByID(id, domainVal, ipVal, createHostRequest.Name, dataCred, dataRapo)
 
-	hosts, err := h.hostService.GetHostsByTenantID(getHostByTenantIDRequest.TenantID)
+	if err != nil {
+
+		return api.WriteJSON(w, http.StatusInternalServerError, err.Error())
+	}
+
+	return api.WriteJSON(w, http.StatusCreated, constructResponse(host))
+}
+
+func (h *HostHandlers) DeleteHostByID(w http.ResponseWriter, req *http.Request) error {
+	id := req.PathValue("id")
+	isDeleted, err := h.hostService.DeleteHostByID(id)
 
 	if err != nil {
 		return api.WriteJSON(w, http.StatusInternalServerError, err.Error())
 	}
 
-	return api.WriteJSON(w, http.StatusOK, hosts)
+	result := make(map[string]string)
+	if isDeleted {
+		result["deleted"] = "true"
+	} else {
+		result["deleted"] = "false"
+	}
+	return api.WriteJSON(w, http.StatusOK, result)
+}
+
+func constructHostForDB(createHostRequest *CreateHostRequest, req *http.Request, h *HostHandlers) *domain.Host {
+	domainVal, ipVal := getDomainIPValues(createHostRequest, h)
+	dataCred, _ := json.Marshal(createHostRequest.Credentials)
+	dataRapo, _ := json.Marshal(createHostRequest.Rapporteurs)
+	tenantID := req.Context().Value(middleware.ContextTenantID)
+	operatorID := req.Context().Value(middleware.ContextUserID)
+
+	host := domain.NewHost(domainVal, ipVal, tenantID.(string), operatorID.(string), createHostRequest.Name, dataCred, dataRapo)
+	return host
+}
+
+func constructResponse(host *domain.Host) *domain.HostResponse {
+	hostResponse := new(domain.HostResponse)
+	hostResponse.Name = host.Name
+	hostResponse.CreatedAt = host.CreatedAt
+	hostResponse.ID = host.ID
+	hostResponse.Domain = host.Domain
+	hostResponse.Ip = host.Ip
+	json.Unmarshal(host.Credentials, &hostResponse.Credentials)
+	json.Unmarshal(host.Rapporteurs, &hostResponse.Rapporteurs)
+	return hostResponse
+}
+
+func getDomainIPValues(createHostRequest *CreateHostRequest, h *HostHandlers) (string, string) {
```

Please correct me if I'm wrong, but I see you're using this method to **fill out the IP attribute** within a host, which is something we should not be doing. We want our users to either pass a Domain OR an IP, and leave it there. 

Later we want to check if this Domain OR IP is valid, which can be done via ping request to either the Domain or the IP in question. 

Additionally, this function should return an error if it fails. You're ignoring that error in line 153

---

It remains in the code but not enable in the api.